libs/content/workspace/src/task_manager.rs

```diff
@@ -457,30 +649,87 @@ impl TaskManager {
                     let in_progress_sync = tasks
                         .in_progress_sync
                         .take()
-                        .expect("Failed to find in-progress entry for sync that just completed");
+                        .expect("failed to find in-progress entry for sync that just completed");
                     // ^ above error may indicate concurrent syncs, which would cause problems
 
-                    let completed_sync = CompletedSync {
-                        status_result,
-                        timing: CompletedTiming::new(in_progress_sync.timing),
-                    };
-                    let in_progress_time = completed_sync
-                        .timing
-                        .started_at
-                        .duration_since(completed_sync.timing.queued_at);
-                    if in_progress_time > Duration::from_secs(1) {
-                        warn!("Sync took {}s", in_progress_time.as_secs());
+                    let timing = CompletedTiming::new(in_progress_sync.timing);
+                    let in_progress_time = timing.completed_at.duration_since(timing.started_at);
+                    if let Err(err) = &status_result {
+                        error!("sync failed ({}ms): {:?}", in_progress_time.as_millis(), err);
+                    } else if in_progress_time > Duration::from_secs(1) {
```

Comment:
```suggestion
                    } else if in_progress_time > Duration::from_secs(5) {
```

my gut on what I would find "alarming", but even still people have all sorts of networks 🤷